File: DSA-DAILY/01-08-2025/118. Pascal's Triangle@leetcode-easy.java
import java.util.*;
class Solution {
    public List<List<Integer>> generate(int num) {
        List<List<Integer>> dp = new ArrayList<>();

        dp.add(new ArrayList<>());
        dp.get(0).add(1);
        for(int i=1;i<num;i++){
            List<Integer> curr = new ArrayList<>();
            List<Integer> prev=dp.get(i-1);
            curr.add(1);
            for(int j=1;j<i;j++){
                int k=prev.get(j-1)+prev.get(j);
                curr.add(k);
            }
            curr.add(1);
            dp.add(curr);
        }
        return dp;
    }
    public static void main(String[] args) {
        Solution sol = new Solution();
        Scanner scanner = new Scanner(System.in);
        int n;
        n=scanner.nextInt();
        List<List<Integer>> result = sol.generate(n);
        for (List<Integer> row : result) {
            System.out.println(row);
        }
    }
}

File: DSA-DAILY/01-08-2025/118. Pascal's Triangle@leetcode-easy.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<vector<int>> generate(int num) {
        int i=0;
        vector<vector<int>> dp;
        dp.push_back({1});
        for(int i=1;i<num;i++){
            vector<int> curr;
            vector<int>& prev=dp[i-1];
            curr.push_back(1);
            for(int j=1;j<i;j++){
                curr.push_back(prev[j-1]+prev[j]);
            }
            curr.push_back(1);
            dp.push_back(curr);
        }
        return dp;
    }
};

int main(){
    Solution sol;
    int n;
    cin>>n;
    vector<vector<int>> result = sol.generate(n);
    for (const auto& row : result) {
        for (int num : row) {
            cout << num << " ";
        }
        cout << endl;
    }
    return 0;
}